src/blockchain/log-processors/order-filled.ts

```diff
@@ -84,25 +84,46 @@ export function calculateProfitLossInOutcome(augur: Augur, trx: Knex.Transaction
   });
 }
 
-export function upsertPositionInMarket(db: Knex, augur: Augur, trx: Knex.Transaction, account: Address, marketID: Address, positionInMarket: Array<string>, callback: ErrorCallback): void {
+export function upsertPositionInMarket(db: Knex, augur: Augur, trx: Knex.Transaction, account: Address, marketID: Address, numTicks: string|number, positionInMarket: Array<string>, callback: ErrorCallback): void {
   trx.select("outcome").from("positions").where({ account, marketID }).asCallback((err: Error|null, positionsRows?: Array<PositionsRow>): void => {
     if (err) return callback(err);
     const numOutcomes = positionInMarket.length;
     const realizedProfitLoss = new Array(numOutcomes);
     const unrealizedProfitLoss = new Array(numOutcomes);
     const positionInMarketAdjustedForUserIntention = new Array(numOutcomes);
-    forEachOf(positionInMarket, (numShares: string, outcome: number, nextOutcome: AsyncCallback): void => {
-      calculateProfitLossInOutcome(augur, trx, account, marketID, outcome, (err: Error|null, profitLossInOutcome?: CalculatedProfitLoss): void => {
-        if (err) return nextOutcome(err);
-        const { realized, unrealized, position } = profitLossInOutcome!;
-        realizedProfitLoss[outcome] = realized;
-        unrealizedProfitLoss[outcome] = unrealized;
-        positionInMarketAdjustedForUserIntention[outcome] = position;
-        nextOutcome();
-      });
-    }, (err: Error|null): void => {
+    augur.api.Market.getShareToken({ _outcome: 0, tx: { to: marketID } }, (err: Error|null, shareToken: Address): void => {
```

Comment:
We're getting outcome 0's ShareToken's total supply and setting it to the shares outstanding of the market?

Reply:
Yes, because shares can only be issued as complete sets.

Reply:
Right! Makes sense.